### ISSUE TITLE
ath11k/ath12k firmware missing, CONFIG_IRQ_REMAP not defined either

### ISSUE THREAD

**Issue by OdinVex** [home-assistant/operating-system#3621]
### Describe the issue you are experiencing

I've attempted to add the firmware for ath11k and ath12k (to support my 17cb:1107 WiFi 7 card) to `/mnt/data/supervisor/firmware` and reboot but I'm not sure it's loading, dmesg's only output is:

```
[    5.322474] ath12k_pci 0000:06:00.0: BAR 0: assigned [mem 0xf0200000-0xf03fffff 64bit]
[    5.322512] ath12k_pci 0000:06:00.0: enabling device (0100 -> 0102)
[    5.323444] ath12k_pci 0000:06:00.0: failed to get 16 MSI vectors, only -28 available
[    5.323473] ath12k_pci 0000:06:00.0: failed to alloc msi: -28
[    5.325225] ath12k_pci: probe of 0000:06:00.0 failed with error -28
```

### What operating system image do you use?

generic-x86-64 (Generic UEFI capable x86-64 systems)

### What version of Home Assistant Operating System is installed?

6.6.46-haos

### Did the problem occur after upgrading the Operating System?

Ongoing

### Hardware details

17cb:1107, WiFi 7 Card, a Qualcomm "FastConnect" WCN7850 (WCN7851) M.2 M-Key.

### Steps to reproduce the issue

1. Install an ath11k/ath12k wireless card.
2. Boot the system
3. Copy firmware files.
4. Reboot.
...

### Anything in the Supervisor logs that might be useful for us?

```txt
N/A, None
```

### Anything in the Host logs that might be useful for us?

```txt
[    5.322474] ath12k_pci 0000:06:00.0: BAR 0: assigned [mem 0xf0200000-0xf03fffff 64bit]
[    5.322512] ath12k_pci 0000:06:00.0: enabling device (0100 -> 0102)
[    5.323444] ath12k_pci 0000:06:00.0: failed to get 16 MSI vectors, only -28 available
[    5.323473] ath12k_pci 0000:06:00.0: failed to alloc msi: -28
[    5.325225] ath12k_pci: probe of 0000:06:00.0 failed with error -28
```

### Additional information

Permissions on the files are correct, I even tried just copying ath11k and ath12k folders alone since they don't exist in HAOS, no effect. Rebooting server doesn't change anything.

**Comment by agners** [comment]
-28 is `ENOSPC`, and it seems that [CONFIG_IRQ_REMAP](https://www.kernelconfig.io/config_irq_remap) is about IRQ remapping. On my Arch Linux distribution that config is enabled, I think we can safely enable it for HAOS as well.

**Comment by agners** [comment]
> Edit: It'd also be nice to include the ath*k firmwares out-of-box considering they're required for many net-related devices.

Hm, we don't ship any ath11k today, I guess shipping firmwares for the chips which are available as a general purpose PCIe device makes sense. The problem is, things are a bit tight on the root file system partition, so depending on size, it might not be possible to add the firmwares right now. I'll leave this up to @sairon to decide.

**Comment by agners** [comment]
> Thank you @agners for adding the kernel config option. I use the fully-managed HAOS-on-metal install, any guess at when it might reach regular updates for it?

HAOS releases don't follow a fixed pattern. We just released HAOS 13.2, it will be a couple of weeks at least until the next update.

You can update to a dev version however, and it is fairly safe to do so since we have the A/B type fallback. Also, the current code base is pretty much 13.2 plus this change (and a stable kernel bump).

`ha os update --version 13.2.dev20241007`, but you'll have to wait until [the build is finished](https://github.com/home-assistant/operating-system/actions/runs/11447966108).
